src/tests/contract_test/wallet_test.cljs

```diff
@@ -0,0 +1,81 @@
+(ns tests.contract-test.wallet-test
+  (:require
+   [cljs.test :refer [deftest is]]
+   [day8.re-frame.test :as rf-test]
+   legacy.status-im.events
+   [legacy.status-im.multiaccounts.logout.core :as logout]
+   legacy.status-im.subs.root
+   [status-im.common.emoji-picker.utils :as emoji-picker.utils]
+   [status-im.contexts.wallet.data-store :as data-store]
+   status-im.events
+   status-im.navigation.core
+   status-im.subs.root
+   [test-helpers.integration :as h]
+   [tests.contract-test.utils :as contract-utils]))
+
+(defn assert-accounts-get-accounts
+  [result]
+  (is (true? (some :wallet result)))
+  (is (true? (some :chat result)))
+  (is (= 2 (count result))))
+
+(deftest accounts-get-accounts-contract
+  (h/log-headline :contract/accounts-get-accounts)
+  (rf-test/run-test-async
+   (h/with-app-initialized
+     (h/with-account
+       (contract-utils/call-rpc-endpoint
+        {:rpc-endpoint "accounts_getAccounts"
+         :action assert-accounts-get-accounts})
+       (h/logout)
+       (rf-test/wait-for
+        [::logout/logout-method])))))
+
+(defn get-default-account [accounts]
+  (first accounts))
+
+(defn check-emoji-is-updated [test-emoji]
+  (fn [accounts]
+    (let [default-account (first accounts)]
+      (is (= (:emoji default-account) test-emoji)))))
+
+(deftest accounts-get-accounts-contract
+  (h/log-headline :contract/accounts-save-account)
+  (rf-test/run-test-async
+   (h/with-app-initialized
+     (h/with-account
+       (let [test-emoji (emoji-picker.utils/random-emoji)
+             account (contract-utils/call-rpc-endpoint
+                      {:rpc-endpoint "accounts_getAccounts"
+                       :action get-default-account})]
+         (contract-utils/call-rpc-endpoint
+          {:rpc-endpoint "accounts_saveAccount"
+           :action identity
+           :params [(data-store/<-account (merge account {:emoji test-emoji}))]})
+         (contract-utils/call-rpc-endpoint
+          {:rpc-endpoint "accounts_getAccounts"
+           :action (check-emoji-is-updated test-emoji)})
+         (h/logout)
+         (rf-test/wait-for
+          [::logout/logout-method]))))))
+
+(defn assert-ethereum-chains [response]
```

Comment:
@shivekkhurana & @ulisesmac both suggested using malli to spec the response. 
I think this is a good idea but in all cases we don't need this because we might only care about certain parts of the response, allowing more flexibility for the Status-Go devs to make "safer" breaking changes.
However it would still be good to have the option as probably there are some endpoints we need to be stricter on maintaining the exact same response. 
We can experiment with the idea to make sure there is sufficient support to use Malli for these tests where needed.

Reply:
That'd be great guys! The effort would be in maintaining the schemas, but those would be able to be reused throughout the project in event handlers and partially in subscriptions, so it pays off for sure.

The opposite is even nicer, we should ideally use malli in re-frame events calling endpoints (also in subs), and then, from extension, when we test those endpoints as contract tests, the schemas would be ready to be used. Imagine how easier it would be to know what are the things circulating through all subs and events...

A recent example from an `events.cljs` file where the schema for the RPC response is defined https://github.com/status-im/status-mobile/pull/18636/files#diff-cadbfd7f815eb76b839764473e1626dbeec7745c0844ac4983637609fedd12aaR37-R45

```clojure
(def ^:private ?permissioned-balances-response
  [:map-of
   ?account-address
   [:sequential
    [:map {:closed true}
     [:type int?]
     [:symbol string?]
     [:decimals int?]
     [:amount string?]]]])
```